Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #67875
* #67874
* #67721
* #66980
* #66979
* #67216
* __->__ #66914
* #66913

Differential Revision: [D31856105](https://our.internmc.facebook.com/intern/diff/D31856105)

peterbell10 committed:
Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}"


Differential Revision: [D31856105](https://our.internmc.facebook.com/intern/diff/D31856105)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}"


Differential Revision: [D31856105](https://our.internmc.facebook.com/intern/diff/D31856105)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}"


Differential Revision: [D31856105](https://our.internmc.facebook.com/intern/diff/D31856105)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856105).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}"


Differential Revision: [D31856105](https://our.internmc.facebook.com/intern/diff/D31856105)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}"


Differential Revision: [D31856105](https://our.internmc.facebook.com/intern/diff/D31856105)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856105).

Commit:
Update on "Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}"


Differential Revision: [D31856105](https://our.internmc.facebook.com/intern/diff/D31856105)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}"


Differential Revision: [D31856105](https://our.internmc.facebook.com/intern/diff/D31856105)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}"


Differential Revision: [D31856105](https://our.internmc.facebook.com/intern/diff/D31856105)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}"


Differential Revision: [D31856105](https://our.internmc.facebook.com/intern/diff/D31856105)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856105).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}"


Differential Revision: [D31856105](https://our.internmc.facebook.com/intern/diff/D31856105)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856105).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}"


Differential Revision: [D31856105](https://our.internmc.facebook.com/intern/diff/D31856105)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}"


Differential Revision: [D31856105](https://our.internmc.facebook.com/intern/diff/D31856105)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856105).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}"


Differential Revision: [D31856105](https://our.internmc.facebook.com/intern/diff/D31856105)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}"


Differential Revision: [D31856105](https://our.internmc.facebook.com/intern/diff/D31856105)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856105).

dagitses commented on aten/src/ATen/native/cpu/IndexKernel.cpp:
```diff
@@ -116,9 +118,9 @@ struct IndexToOffset {
-  IndexToOffset(const Tensor & tensor) : sizes(tensor.sizes()),
-                                         strides(tensor.strides()),
-                                         ndim(tensor.dim()) {}
+  IndexToOffset(const TensorBase & tensor) : sizes(tensor.sizes()),
```
internal lint is complaining about this needing to be marked explicit. I know it's not your code but might as well make it happy.

dagitses commented on aten/src/ATen/native/cpu/IndexKernel.cpp:
```diff
@@ -116,9 +118,9 @@ struct IndexToOffset {
-                                         ndim(tensor.dim()) {}
+  IndexToOffset(const TensorBase & tensor) : sizes(tensor.sizes()),
+                                             strides(tensor.strides()),
+                                             ndim(tensor.dim()) {}
```
internal lint is complaining about narrowing from 64-bit to 32-bit here. Make ndim member 64-bit?

dagitses commented on aten/src/ATen/native/cuda/IndexKernel.cpp:
```diff
@@ -0,0 +1,73 @@
+  // owning and expand_outplace returns a borrow, the returned borrow
+  // would dangle.
+  auto mask_self_expanded = expand_outplace(*mask_temp, *self_temp);
+  at::native::index_out(result, *std::get<1>(mask_self_expanded), c10::List<c10::optional<at::Tensor>>({*std::get<0>(std::move(mask_self_expanded))}));
```
from internal lint:
> Potential variable access after move.
> Variable mask_self_expanded is accessed here, and is moved-from in an unspecified order

dagitses replied:
i see a formatting improvement, but I don't see any material changes here.

peterbell10 replied:
The `std::move` now comes after the `std::get`, so only one tuple element is created as an r-value reference.

dagitses replied:
ah thanks, i see that now.

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}"


Differential Revision: [D31856105](https://our.internmc.facebook.com/intern/diff/D31856105)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856105).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}"


Differential Revision: [D31856105](https://our.internmc.facebook.com/intern/diff/D31856105)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856105).

Commit:
Update on "Remove native_functions.yaml dependency from IndexKernel.{cpp,cu}"


Differential Revision: [D31856105](https://our.internmc.facebook.com/intern/diff/D31856105)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856105).
